v22.2/recommended-production-settings.md

```diff
@@ -343,6 +343,10 @@ Each node has a default SQL memory size of `25%`. This memory is used as-needed
     SQL memory size applies a limit globally to all sessions at any point in time. Certain disk-spilling operations also respect a memory limit that applies locally to a single operation within a single query. This limit is configured via a separate cluster setting. For details, see [Disk-spilling operations](vectorized-execution.html#disk-spilling-operations).
     {{site.data.alerts.end}}
 
+    If a node runs out of its allocated SQL memory, a `memory budget exceeded` error occurs. To mitigate this issue, increase `--max-sql-memory` on the node.
+
+    Do not set `--max-sql-memory` too high. If it reaches the total memory available to the `cockroach` process, the Go runtime’s invisible memory allocations are not accounted for and will cause the process to crash or to become very slow by going to disk swap. Try different values and monitor your cluster's performance. Avoid increasing the value further as soon the total memory usage under load grows beyond 80% of overall capacity available to the process. If you are still encountering `memory budget exceeded` errors, consider increasing the amount of RAM on the node. For more information, refer to the CockroachDB blog post [Memory usage in CockroachDB](https://cockroachlabs.com/blog/memory-usage-cockroachdb/).
```

Comment:
If we're being very precise, "Go runtime's invisible memory allocations" is not exactly correct - the Go runtime is always able to keep track of all allocations, but it's the memory accounting system that is built into CRDB that attempts to respect `--max-sql-memory` limit, and the allocations are invisible from CRDB's accounting system. (The reason is that this system is opt in, i.e. an engineer must write additional code to register all of the memory allocations with the CRDB's memory accounting system.)

I'd probably just drop some details from the second sentence in the paragraph, to something like "If it reaches the total memory available to the `cockroach` process, then the process is likely to crash or to become very slow ...". WDYT?

Reply:
Thanks for this, I copied that info directly from the linked blog (with some very light copyediting).